js/index.js

```diff
@@ -1,7 +1,75 @@
+  
+  }
+  else { console.log(`Wow, you both have equally long names, ${hacker1.length} characters`)
+  }
+  
+
+// Iteration 3: Loops
+let driverUppercaseName = "";
+for (let i = 0; i < hacker1.length ; i++) {
+  driverUppercaseName += hacker1[i].toUpperCase() + " ";
+ }
+ console.log(driverUppercaseName)
+
+let navigatorInverseName= ""
+ for (let i = hacker2.length-1; i >= 0  ; i--) {
+  navigatorInverseName += hacker2[i] ;
+ }
+ console.log(navigatorInverseName)
+
+const namesOrder = hacker1.localeCompare(hacker2)
+
+
+   if (namesOrder === -1) { 
+        console.log (`The driver's name goes first.`)}
+     else if (namesOrder === 1) {
+       console.log (`Yo, the navigator goes first definitely.`)}
+     else {
+       console.log(`What?! You both have the same name?`)
+     }
+ 
+//Bonus 1
+
+const loremIpsum = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent sed mi facilisis, vulputate nunc id, aliquam diam. Integer elit diam, rutrum convallis felis non, egestas tincidunt tellus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Duis feugiat tincidunt lacus non accumsan. In hac habitasse platea dictumst. Morbi ipsum leo, tempor vel mattis vel, lobortis non diam. Pellentesque ac nisi pulvinar, pellentesque odio pellentesque, consequat massa. Maecenas at lectus orci. Aenean aliquam tempor ex maximus tincidunt. Nunc tempor auctor est tincidunt rutrum. Suspendisse porttitor mi risus, sit amet tempus mi accumsan a. Phasellus efficitur magna sed porttitor pulvinar. Aliquam vel bibendum nulla, sed fringilla eros. Cras convallis consequat iaculis. Sed quis ligula lorem. Pellentesque id lobortis elit, et fermentum quam. Nam sagittis id erat eget condimentum. Morbi pharetra sollicitudin tristique. Proin metus diam, imperdiet sed lacinia quis, lacinia at nisi. Ut a nulla quis ex tempus tincidunt. Ut sit amet feugiat nunc. Vestibulum cursus mauris est, ut convallis lacus blandit vel. Proin id varius libero. Aenean id risus vestibulum erat fringilla fermentum. Phasellus lobortis risus ac dui laoreet bibendum. Morbi volutpat sapien est, semper efficitur mauris feugiat nec. Nam tincidunt rutrum ipsum, id laoreet nulla malesuada tincidunt. Nam ut lorem est. Donec ultrices at elit id tristique. Phasellus luctus diam orci, dignissim tincidunt nunc fermentum sed. Sed vitae egestas est, eget tempus ipsum. Aliquam lacus sem, auctor quis dui a, interdum ullamcorper ante. Cras gravida convallis urna, sit amet auctor leo facilisis in. Integer sollicitudin gravida diam non tempus. Nam quis tempor nibh, scelerisque commodo nibh. In hac habitasse platea dictumst."
+
+let wordsNumber = 0
+for (i = 0; i < loremIpsum.length; i++){
+ if (loremIpsum[i] != " "){
+   wordsNumber ++
+ }
+
+}
+
+console.log(wordsNumber)
+
+let countWordEt = 0
+for (i = 0; i < loremIpsum.length; i++){
+ if (loremIpsum[i] === "et") {
+    countWordEt ++
+  }
+}
+console.log(countWordEt)
+
+//Bonus2
+const palindrome ="step on no pets"
```

Comment:
Bien!, pero qué pasa si hay signos de puntuación, mayúsculas, exclamaciones... en la frase a testear?

Para aplicar esta lógica podemos tener en cuenta DOS casos.

Uno en el que no hay signos de puntuación, exclamación, ni caractéres raros que comparar... en ese caso, es coger nuestro string, unirla y ver si a la inveresa es igual.

Dos, el caso complejo que es el que habéis tratado de hacer, que es cuando tenemos signos de puntuación o caracteres extraños:

1. Creo una variable que sea igual a lo que escribamos en el prompt, quitándole todos los caracteres raros, para hacer esto puedo usar la función replace, pasando una REGEX, que es este string ` /[\s,.!¡¿?:;']/g, ''` que lo hace es buscar caracteres como `!,.,¿,?,:,...` y los reemplaza por `string` vacío `''` así conseguimos que por ejemplo `!ana!, ana?`, se nos quede en la palabra o frase que queremos comprobar.
También va concatenada otra regex que lo que hace es quitarle los acentos a las palabras, para que también no intercedan a la hora de comprobarlo. Después de limpiar la frase de este tipo de caracteres lo pasamos todo a minúsculas para que también no intercedan las mayúsculas a la hora de comparar. Esto además quitarría también los espacios en blanco. Quedaría así:

Por ejemplo la frase: `A mamá Roma, le aviva el amor a papá y a papá Roma le aviva, el amor a mamá.`

```
const isPalindrome = prompt("Let's write a palindrome, we will check it!").replace(/[\s,.!¡¿?:;']/g, '').normalize("NFD").replace(/[\u0300-\u036f]/g, "").toLowerCase()
```

Esto daría: `amamaromaleavivaelamorapapayapaparomaleavivaelamoramama`

2. Ahora solo quedaría comprobar que esta frase y su invertida son iguales:

```
  if (isPalindrome == isPalindrome.split('').reverse().join('')){
   console.log('The sentence you wrote, is a palindrome!' + '\n')
 } else {
   console.log('Sorry its not a palindrome!' + '\n')
 }
```